### PR TITLE
Manaphy Egg Quest

### PR DESCRIPTION
**Pull request by Leafsw0rd**
A Manaphy questline based in Sinnoh. This questline incorporates the Manaphy Egg quest lines from the three original ranger games - primarily the first and an... 'upscaled' version of the second. 

This quest can be started any time after getting the Grass Gym Badge, but can only be completed once the player can reach Canalave and Pal Park. Roaming Manaphy is locked behind this questline - because Manaphy could be obtained at any time in Sinnoh before this update occurs, this quest guarantees a Manaphy upon quest completion. Trading a chance for an early Manaphy for a guaranteed later Manaphy.

This quest has - by my understanding - the first case where a Dungeon boss disappears and then reappears so far (as opposed to doing one or the other, like the Sealed Chamber, or appearing then disappearing, like Vivillon.) It's not anything revolutionary, it's just a MultiRequirement, but it helps sell the questline's plot. 

At time of writing, two placeholder images are still in place. Only one is really disruptive to the questline, the Strange Boulders placeholder. 

The Beachcomber's dialogue has also been changed to reflect the updated requirements and storyline.

### CODE REVIEW

**Comment by DraKay** [comment]
You haven't eddit credits.ts file

**Comment by Leafsw0rd** [comment]
After a few stumbles under the hood, this should be a version with much improved Quest Text in terms of informativeness, the remaining placeholders exchanged, and credits properly given.

File: src/scripts/towns/TownList.ts [hunk 1]
```diff
@@ -2864,6 +2867,166 @@ const LucyStevens1 = new NPC('Lucy Stevens', [
...
+    '<i>Perhaps these aren\'t boulders after all... and if they are what you think they are, a good soaking should settle the matter.</i>',
+], {
+    image: 'assets/images/npcs/Strange Boulders.png',
+    requirement: new MultiRequirement([new QuestLineStepCompletedRequirement('Recover the Precious Egg!', 0), new QuestLineStepCompletedRequirement('Recover the Precious Egg!', 2, GameConstants.AchievementOption.less)]),
+});
+
+const ManaphyGoRockCommander = new NPC('Go-Rock Commander', [
+    'You... are you a Pokémon Ranger? ...No? Well, no matter.',
+    'Allow me to explain something to you. This egg is meaningful only if we possess it.',
+    'We\'re going to hatch Manaphy from this egg and get it to guide us to it\'s Sea Temple.',
```
Comment (Aegyo) [inline src/scripts/towns/TownList.ts:2921]:
```suggestion
    'We\'re going to hatch Manaphy from this egg and get it to guide us to its Sea Temple.',
```

File: src/scripts/towns/TownList.ts [hunk 2]
```diff
@@ -2864,6 +2867,166 @@ const LucyStevens1 = new NPC('Lucy Stevens', [
...
+    'Ah, you\'re the one chasing after that Happiny? You look a bit puffed, are you okay?',
+    'I tried to capture it, but it barrelled right through my Capture Disc\'s trail! It went for Hearthome City. It wasn\'t showing any signs of slowing down yet.',
+], {
+    image: 'assets/images/npcs/Pokemon Ranger (female).png',
+    requirement: new MultiRequirement([new QuestLineStepCompletedRequirement('Recover the Precious Egg!', 16), new QuestLineStepCompletedRequirement('Recover the Precious Egg!', 18, GameConstants.AchievementOption.less)]),
+});
+
+const HappinyWitness6 = new NPC('Murph', [
+    'Me and my buddy Slowpoke are trying to catch a Happiny that just went running into Mt. Coronet.',
+    'It had a blue egg on it\'s belly - I\'ve definitely seen an egg like that before, back in Fiore.',
```
Comment (Aegyo) [inline src/scripts/towns/TownList.ts:2986]:
```suggestion
    'It had a blue egg on its belly - I\'ve definitely seen an egg like that before, back in Fiore.',
```

File: src/scripts/towns/TownList.ts [hunk 3]
```diff
@@ -2864,6 +2867,166 @@ const LucyStevens1 = new NPC('Lucy Stevens', [
...
+    'Some guy in a weird white outfit was chasing a Happiny!',
+    'They were heading south, towards Professor Rowan\'s lab!',
+    'You\'re chasing after them too, right? Good luck catching up to them!',
+], {
+    image: 'assets/images/trainers/Bug Catcher.png',
+    requirement: new MultiRequirement([new QuestLineStepCompletedRequirement('Recover the Precious Egg!', 19), new QuestLineStepCompletedRequirement('Recover the Precious Egg!', 21, GameConstants.AchievementOption.less)]),
+});
+
+const HappinyWitness9 = new NPC('Go-Rock Pincher', [
+    'Damn, that Happiny got away. Those little friends got me good.',
```
Comment (Aegyo) [inline src/scripts/towns/TownList.ts:3013]: What are "Those little friends"?

**Review by Aegyo — CHANGES_REQUESTED** [review]
Quest works, and the temp battle statistics update looks correct too.
Just a couple of text fixes.

File: src/scripts/towns/TownList.ts (hunk 3 above)
Comment (Leafsw0rd) [reply src/scripts/towns/TownList.ts:3013]: "those little friends" are the Pichu and Glameow that join Happiny. THey're from the Shadows of Almia questline. 
I'll reword the quest to have them explained a bit more clearly.

**Review by Aegyo — APPROVED** [review]

**Pull request merged**